### PR TITLE
Fix identity fuse for oneDNN

### PR DESCRIPTION
**Pull request by bgawrych**
## Description ##
During testing https://github.com/dmlc/gluon-nlp/pull/1581 it occurred that identity fuse doesn't work properly on big graph, but works in unit test - that was caused by lack of stop condition in subgraph selector.

This PR fix this issue and expand test to cover such cases

### CODE REVIEW

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-gpu, unix-cpu, unix-gpu]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-gpu, unix-cpu]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-gpu]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-gpu]

File: src/operator/subgraph/dnnl/dnnl_identity_property.h [hunk 1]
```diff
@@ -65,10 +66,11 @@ class SgDNNLIdentitySelector : public SubgraphSelectorV2 {
   }
 
   bool SelectInput(const BiDirectedNode& n, const BiDirectedNode& input_node) override {
-    if (input_node.node->is_variable()) {
+    if (pattern_found || input_node.node->is_variable()) {
```
Comment (PawelGlomski-Intel) [inline src/operator/subgraph/dnnl/dnnl_identity_property.h:69]: You could use `matched_list_` like you did in the `Filter` function.
```suggestion
    if (matched_list_.size() == 2 || input_node.node->is_variable()) {
```
If not, then I would also use `pattern_found` in the `Filter` function, to show, that these conditions are connected.
https://github.com/apache/incubator-mxnet/blob/7e163157cf789d7ec5e7272532e04495e57bd0ba/src/operator/subgraph/dnnl/dnnl_identity_property.h#L85
Change to :
```
    if (pattern_found && candidates.size() == matched_list_.size()) { 
```

**Review by PawelGlomski-Intel — APPROVED** [review]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [all]

File: src/operator/subgraph/dnnl/dnnl_identity_property.h (hunk 1 above)
Comment (bgawrych) [reply src/operator/subgraph/dnnl/dnnl_identity_property.h:69]: Done

**Review by agrabows — APPROVED** [review]

**Review by bartekkuncer — APPROVED** [review]
Looks good.

**Pull request merged**